src/endpoint/s3/ops/s3_get_object.js

```diff
@@ -52,42 +52,11 @@ async function get_object(req, res) {
     if (md_params.get_from_cache) {
         params.get_from_cache = true;
     }
-    if (part_number) {
-        params.part_number = part_number;
-    }
-    try {
-        const ranges = http_utils.normalize_http_ranges(
-            http_utils.parse_http_ranges(req.headers.range),
-            obj_size
-        );
-        if (ranges) {
-            // reply with HTTP 206 Partial Content
-            res.statusCode = 206;
-            params.start = ranges[0].start;
-            params.end = ranges[0].end;
-            const content_range = `bytes ${params.start}-${params.end - 1}/${obj_size}`;
-            dbg.log1('reading object range', req.path, content_range, ranges);
-            res.setHeader('Content-Range', content_range);
-            res.setHeader('Content-Length', params.end - params.start);
-            // res.header('Cache-Control', 'max-age=0' || 'no-cache');
-        } else {
-            dbg.log1('reading object', req.path, obj_size);
-        }
-    } catch (err) {
-        if (err.ranges_code === 400) {
-            // return http 400 Bad Request
-            dbg.log1('bad range request', req.headers.range, req.path, obj_size);
-            throw new S3Error(S3Error.InvalidArgument);
-        }
-        if (err.ranges_code === 416) {
-            // return http 416 Requested Range Not Satisfiable
-            dbg.warn('invalid range', req.headers.range, req.path, obj_size);
-            // let the client know of the relevant range
-            res.setHeader('Content-Range', 'bytes */' + obj_size);
-            throw new S3Error(S3Error.InvalidRange);
-        }
-        throw err;
+    if (multipart_number) {
+        params.multipart_number = multipart_number;
     }
+    s3_utils.s3_range(req, res, object_md, obj_size, params);
```

Comment:
This function name is neither imperative nor declarative.
I would rename to something like `handle_range` or `set_response_range`.